Add flag to end regression after N failures (#5309)
cocotb/cocotb: docs/source/library_reference.rst

Discovering Tests
=================

.. envvar:: COCOTB_MAX_FAILURES

    Type: :ref:`_env-boolean`

    Limits the number of test failures allowed during a regression run.

    If set then if the number of failed tests reaches this value, the regression
    is terminated early and no further tests are executed.

    If not set this disables early termination and causes

>     all tests to be executed regardless of failures.

Reviewer: ```suggestion
    If not set, all tests are executed regardless of failures.
```
No information loss with the shorter sentence, right?
